src/pool/cli.c

```diff
@@ -1524,128 +1524,10 @@ struct pool_evict_state {
-pool_evict_cp(tse_task_t *task, void *data)
-{
-	struct pool_evict_state	*state = dc_task_get_priv(task);
-	crt_rpc_t		*rpc = *((crt_rpc_t **)data);
-	struct pool_evict_in	*in = crt_req_get(rpc);
-	struct pool_evict_out	*out = crt_reply_get(rpc);
-	bool			 free_state = true;
-	int			 rc = task->dt_result;
-
-	rc = rsvc_client_complete_rpc(&state->client, &rpc->cr_ep, rc,
-				      out->pvo_op.po_rc, &out->pvo_op.po_hint);
-	if (rc == RSVC_CLIENT_RECHOOSE ||
-	    (rc == RSVC_CLIENT_PROCEED &&
-	     daos_rpc_retryable_rc(out->pvo_op.po_rc))) {
-		rc = tse_task_reinit(task);
-		if (rc != 0)
-			D_GOTO(out, rc);
-		free_state = false;
-		D_GOTO(out, rc = 0);
-	}
-
-	if (rc != 0) {
-		D_ERROR("RPC error while evicting pool handles: "DF_RC"\n",
-			DP_RC(rc));
-		D_GOTO(out, rc);
-	}
-
-	rc = out->pvo_op.po_rc;
-	if (rc != 0) {
-		D_ERROR("failed to evict pool handles: "DF_RC"\n", DP_RC(rc));
-		D_GOTO(out, rc);
-	}
-
-	D_DEBUG(DF_DSMC, DF_UUID": evicted\n", DP_UUID(in->pvi_op.pi_uuid));
-
-out:
-	crt_req_decref(rpc);
-	if (free_state) {
-		rsvc_client_fini(&state->client);
-		dc_mgmt_sys_detach(state->sys);
-		D_FREE(state);
-	}
-	return rc;
-}
-
 int
-dc_pool_evict(tse_task_t *task)
+dc_pool_dummy(tse_task_t *task)
 {
-	struct pool_evict_state	*state;
-	crt_endpoint_t		 ep;
-	daos_pool_evict_t	*args;
-	crt_rpc_t		*rpc;
-	struct pool_evict_in	*in;
-	int			 rc;
-
-	args = dc_task_get_args(task);
-	state = dc_task_get_priv(task);
-
-	if (state == NULL) {
-		if (!daos_uuid_valid(args->uuid))
-			D_GOTO(out_task, rc = -DER_INVAL);
-
-		D_DEBUG(DF_DSMC, DF_UUID": evicting\n", DP_UUID(args->uuid));
-
-		D_ALLOC_PTR(state);
-		if (state == NULL) {
-			D_GOTO(out_task, rc = -DER_NOMEM);
-		}
-
-		rc = dc_mgmt_sys_attach(args->grp, &state->sys);
-		if (rc != 0)
-			D_GOTO(out_state, rc);
-		rc = rsvc_client_init(&state->client, args->svc);
-		if (rc != 0)
-			D_GOTO(out_group, rc);
-
-		daos_task_set_priv(task, state);
-	}
-
-	ep.ep_grp = state->sys->sy_group;
-	rc = dc_pool_choose_svc_rank(args->uuid, &state->client,
-				     NULL /* mutex */, state->sys, &ep);
-	if (rc != 0) {
-		D_ERROR(DF_UUID": cannot find pool service: "DF_RC"\n",
-			DP_UUID(args->uuid), DP_RC(rc));
-		goto out_client;
-	}
-	rc = pool_req_create(daos_task2ctx(task), &ep, POOL_EVICT, &rpc);
-	if (rc != 0) {
-		D_ERROR(DF_UUID": failed to create pool evict rpc: %d\n",
-			DP_UUID(args->uuid), rc);
-		D_GOTO(out_client, rc);
-	}
-
-	in = crt_req_get(rpc);
-	uuid_copy(in->pvi_op.pi_uuid, args->uuid);
-
-	crt_req_addref(rpc);
-
-	rc = tse_task_register_comp_cb(task, pool_evict_cp, &rpc, sizeof(rpc));
-	if (rc != 0)
-		D_GOTO(out_rpc, rc);
-
-	rc = daos_rpc_send(rpc, task);
-	if (rc != 0)
-		D_GOTO(out_rpc, rc);
-
-	return rc;
-
-out_rpc:
-	crt_req_decref(rpc);
-	crt_req_decref(rpc);
-out_client:
-	rsvc_client_fini(&state->client);
-out_group:
-	dc_mgmt_sys_detach(state->sys);
-out_state:
-	D_FREE(state);
-out_task:
-	tse_task_complete(task, rc);
-	return rc;
+	return -DER_NOSYS;
```

Comment:
I wonder if we need to invoke task_complete here ?

Either way, in @GitHuaKuang 's patch, I asked him to name it dc_deprecated and maybe print a warning or error.

Reply:
We should be able to share the same dc_deprecated() in my PR-3984, no need for each individual functions.

Reply:
@GitHuaKuang, I agree, I need to wait for your changes to be landed or merge your changes to my PR.